mission_control/test/test_attitude_servo_behavior.py

```diff
@@ -115,9 +118,12 @@ def attitude_servo_goals_are_set():
         self.simulated_auv_interface_data_pub.publish(msg)
 
         def success_mission_status_is_reported():
-            return self.mission.execute_mission_state == ReportExecuteMissionState.COMPLETE
+            if self.mission.execute_mission_state == ReportExecuteMissionState.ABORTING:
+                self.mission_has_failed = True
+            return (self.mission.execute_mission_state == ReportExecuteMissionState.COMPLETE and
+                    not self.mission_has_failed)
```

Comment:
@patapufate +1 to check for state transitions, but this check is flaky. It doesn't check that the sequence of states was correct, and it races with mission control to detect the temporary ABORTING state. 

I'd suggest we only check to final states OR that we do checks on a list of all reported states so far.

Reply:
@hidmic I have implemented only in altitude behavior test before move to all. PTAL a84d649